mozillaspeechlibrary/src/main/java/com/mozilla/speechlibrary/networking/GeckoNetworking.java

```diff
@@ -0,0 +1,94 @@
+package com.mozilla.speechlibrary.networking;
+
+import android.content.Context;
+import android.os.Handler;
+import android.os.Looper;
+
+import com.mozilla.speechlibrary.MozillaSpeechService;
+import com.mozilla.speechlibrary.STTResult;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.mozilla.geckoview.GeckoRuntime;
+import org.mozilla.geckoview.GeckoWebExecutor;
+import org.mozilla.geckoview.WebRequest;
+
+import java.io.ByteArrayOutputStream;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+
+public class GeckoNetworking extends Networking {
+
+    private Handler mHandler;
+
+    public GeckoNetworking(Context aContext, MozillaSpeechService aSpeechService) {
+        super(aContext, aSpeechService);
+
+        mHandler = new Handler(Looper.getMainLooper());
+    }
+
+    @Override
+    public void doSTT(final ByteArrayOutputStream baos, final NetworkSettings mNetworkSettings) {
+        if (cancelled) {
+            mSpeechService.notifyListeners(MozillaSpeechService.SpeechState.CANCELED, null);
+            return;
+        }
+
+        mHandler.post(() -> {
+            GeckoWebExecutor executor = new GeckoWebExecutor(GeckoRuntime.getDefault(mContext));
+            ByteBuffer input = ByteBuffer.allocateDirect(baos.size());
+            input.put(baos.toByteArray());
+            WebRequest request = new WebRequest.Builder(STT_ENDPOINT)
+                    .body(input)
+                    .addHeader("Accept-Language-STT", mNetworkSettings.mLanguage)
+                    .addHeader("Store-Transcription", mNetworkSettings.mStoreTranscriptions ? "1": "0" )
+                    .addHeader("Store-Sample", mNetworkSettings.mStoreSamples ? "1": "0")
+                    .addHeader("Product-Tag", mNetworkSettings.mProductTag)
+                    .addHeader("Content-Type", "audio/3gpp")
+                    .build();
+
+            mSpeechService.notifyListeners(MozillaSpeechService.SpeechState.DECODING, null);
+            executor.fetch(request).then(webResponse -> {
+                String body;
+                if (webResponse != null) {
+                    if (webResponse.body != null) {
+                        int size = webResponse.body.available();
```

Comment:
But even after applying the changes aforementioned and having confirmed that the json is returned from the server to the client using Wireshark, this always return 0. So seems to me that the executor is not reading the result properly.

Reply:
I from https://docs.oracle.com/javase/7/docs/api/java/io/InputStream.html#available()

> Note that while some implementations of InputStream will return the total number of bytes in the stream, many will not. It is never correct to use the return value of this method to allocate a buffer intended to hold all data in this stream. 

Just call `read()`

Reply:
Moving it to read worked fine. Fixed.